backend/dataall/base/aws/s3_client.py

```diff
@@ -33,3 +34,31 @@ def get_presigned_url(region, bucket, key, expire_minutes: int = 15):
         except ClientError as e:
             log.error(f'Failed to get presigned URL due to: {e}')
             raise e
+
+    @staticmethod
+    def object_exists(region, bucket, key) -> bool:
+        try:
+            S3_client.client(region, None).head_object(Bucket=bucket, Key=key)
+            return True
+        except ClientError as e:
```

Comment:
nit: the error logs here might be a bit misleading:
- 404 Not Found is "ok", we should just return False
- for everything else we could raise an `UnauthorizedOperation` for the action `CHECK_S3_OBJECT_EXISTS` and log the error

Reply:
changes are done, implemented under `worksheets/aws/s3_client.py`